qemu/tests/virtio_driver_sign_check.py

```diff
@@ -1,85 +1,71 @@
 import logging
-import os
-import re
-import time
 
+from virttest import error_context
 from virttest import utils_misc
+from virttest.utils_windows import virtio_win
 
 
+def get_driver_file_path(session, params):
+    """
+    Get tested driver path and drive_letter,
+    Path like "E:\\vioscsi\\2k8\\x86\\vioscsi.cat/sys/inf"
+    or "A:\\i386\\Win2008\\vioscsi.cat".
+    drive_letter like "E:" or "A:".
+
+    :param session: VM session
+    :param params: Dictionary with the test parameters
+    """
+    driver_path = params["tested_driver"]
+    media_type = params["virtio_win_media_type"]
+    get_drive_letter = getattr(virtio_win, "drive_letter_%s" % media_type)
+    drive_letter = get_drive_letter(session)
+    get_product_dirname = getattr(virtio_win,
+                                  "product_dirname_%s" % media_type)
+    guest_name = get_product_dirname(session)
+    get_arch_dirname = getattr(virtio_win, "arch_dirname_%s" % media_type)
+    guest_arch = get_arch_dirname(session)
+    path = ("{letter}\\{driver}\\{name}\\{arch}\\" if media_type == "iso"
+            else "{letter}\\{arch}\\{name}\\{driver}").\
+        format(letter=drive_letter, driver=driver_path,
+               name=guest_name, arch=guest_arch)
+    return drive_letter, path
+
+
+@error_context.context_aware
 def run(test, params, env):
     """
     KVM windows virtio driver signed status check test:
     1) Start a windows guest with virtio driver iso/floppy
-    2) Install windows SDK in guest.
-    3) use SignTool.exe to verify whether block driver digital signed
+    2) Generate a tested driver file list.
+    3) use SignTool.exe to verify whether all drivers digital signed
 
     :param test: QEMU test object
     :param params: Dictionary with the test parameters
     :param env: Dictionary with test environment.
     """
     vm = env.get_vm(params["main_vm"])
     vm.verify_alive()
-
-    def is_sdksetup_finished():
-        s, o = session.cmd_status_output("tasklist | find \"SDK\"")
-        if s:
-            return True
-        else:
-            return False
-
-    timeout = float(params.get("login_timeout", 240))
+    timeout = float(params.get("login_timeout", 360))
```

Comment:
Not suggest to customize `login_timeout` if there is no special needs.

Reply:
thanks for you advise, has updated~